caddyconfig/caddyformatters.go

```diff
@@ -0,0 +1,41 @@
+// Copyright 2015 Matthew Holt and The Caddy Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package caddyconfig
+
+import "fmt"
+
+// Formatter is a type which can format a given configuration file.
+// It returns the the formatted result or an error.
+type Formatter interface {
+	Format(body []byte) []byte
+}
+
+// RegisterFormatter registers a config Formatter with the given name.
+// This should usually be done at init-time.
+func RegisterFormatter(name string, formatter Formatter) error {
+	if _, ok := configFormatters[name]; ok {
+		return fmt.Errorf("%s: already registered", name)
+	}
+	configFormatters[name] = formatter
+	return nil
+}
+
+// GetFormatter returns the Formatter with the given name,
+// or nil if one with that name is not registered.
+func GetFormatter(name string) Formatter {
+	return configFormatters[name]
+}
+
+var configFormatters = make(map[string]Formatter)
```

Comment:
As cool as it would be to have formatters be extensible, I really can't think of a need for formatters beyond just the Caddyfile. I mean, I wouldn't want/expect Caddy to be able to format any other general-purpose config language like NGINX configs or YAML or any other config adapters. That'd be kinda out of scope, right?

If you agree, I think we can simplify things a lot. This file wouldn't be needed. No need to register formatters. No `--formatter` flag. Just call the `Format()` function from the command, since there's only the one.

Reply:
Umm, we could remove it. The first thing I thought about this was we could extend it to JSON for starters easily but I get why we wouldn't like to do that.